src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4158,20 +4178,19 @@ private bool HasTopLevelNullabilityConversion(TypeWithAnnotations source, TypeWi
             bool extensionMethodThisArgument = false,
             Optional<LocalState> stateForLambda = default,
             bool trackMembers = false,
-            Location location = null)
+            Location diagnosticLocationOpt = null)
         {
             Debug.Assert(!trackMembers || !IsConditionalState);
-            Debug.Assert(node != null);
-            Debug.Assert(operandOpt != null || !operandType.HasNullType);
```

Comment:
>Debug.Assert(operandOpt != null || !operandType.HasNullType); [](start = 12, length = 61)

why remove this assertion?

Reply:
`operandOpt` is now `convertedNode`, and can never be null.

---
In reply to: [282228829](https://github.com/dotnet/roslyn/pull/35544#discussion_r282228829) [](ancestors = 282228829)

Reply:
There is also not an equivalent relationship between `conversionOpt` and `operandType`

---
In reply to: [282267136](https://github.com/dotnet/roslyn/pull/35544#discussion_r282267136) [](ancestors = 282267136,282228829)